docs/plugins/github-apps.md

```diff
@@ -0,0 +1,78 @@
+
+It also provides a much clearer and better authorization model as a opposed to
+the OAuth apps and their respective scopes.
+
+## Caveats
+
+- It's not possible to have multiple backstage GitHub Apps installed in the same
+  GitHub organization be managed by Backstage. We currently don't check through
+  all the registered GitHub Apps to see which ones are installed for a
+  particular repository. We just respect global Organization installs right now.
+- App permissions is not managed by Backstage. They're created with some simple
+  default permissions which you are free to change as you need, but you will
+  need to update them in the GitHub web console, not in Backstage right now. The
+  permissions that are defaulted are `metadata:read` and `contents:read`.
+- The created GitHub App is private by default, this is most likely what you
+  want for github.com but it's recommended to make your application public for
+  GitHub Enterprise in order to share application across your GHE organizations.
+
+A GitHub app created with `backstage-cli create-github-app` will have read
+access by default. You have to manually update the GitHub App settings in GitHub
+to grant the app more permissions if needed.
+
+### Using the CLI (public GitHub only)
+
+You can use the `backstage-cli` to create GitHub App' using a manifest file that
+we provide. This gives us a way to automate some of the work required to create
+a GitHub app.
+
+You can read more about the `backstage-cli create-github-app` method
+[here](../cli/commands.md#create-github-app)
+
+Once you've gone through the CLI command, it should produce a `yaml` file in the
+root of the project which you can then use as an `include` in your
+`app-config.yaml`. You can go ahead and skip to
+[here](#including-in-integrations-config) if you've got to this part.
+
+### GitHub Enterprise
+
+You have to create the GitHub Application manually using these
+[instructions](https://docs.github.com/en/free-pro-team@latest/developers/apps/creating-a-github-app)
+as GitHub Enterprise does not support creation of apps from manifests.
+
+Once the application is created you have to generate a private key for the
+application it in a `yaml` file.
+
+The yaml file must include the following information. Please note that the
+indentation for the `privateKey` is required.
+
+```yaml
+appId: 1
+clientId: client id
+clientSecret: client secret
+webhookSecret: webhook secret
+privateKey: |
+  -----BEGIN RSA PRIVATE KEY-----
+  ...Key content...
+  -----END RSA PRIVATE KEY-----
+```
+
+### Including in Integrations Config
+
+Once the credentials are stored in a `yaml` file generated by
+`create-github-app` or manually by following the
+[GitHub Enterprise](#gitHub-enterprise) instructions they can be included in the
+`app-config.yaml` under the integrations section.
+
+```yaml
+integrations:
+  github:
+    - host: github.com
+    apps:
+        - $include: example-backstage-app-credentials.yaml
```

Comment:
Maybe we need to clarify around security here since this is a file that holds highly sensitive information. How would this really work in a production scenario? Or for a user that just wants to do some local dev? Will the file be checked into VCS? How will it be shipped in CI/CD? Etc

Reply:
Let's add a line about this file being sensitive in this doc. 

It's should definitely not be checked into git. The idea would be to add `*-credentials.yaml` to the .gitignore template for `create-app`. How the file is delivered to production is highly dependent on your production environment but there's no problems shipping a yaml file with credentials using Vault or by a ConfigMap hooked up to some secret store.

Reply:
@Rugvip does the above sound sane or have you got other suggestions?

If the above sounds like a plan then we need to alter the file name for create-github-app because it does not append `-credentials` at this point.

Reply:
We already have `*.local.yaml` ignored, so I think it's better to stick to that? You'll really need to include that in a `app-config.local.yaml` config as well, so it's pretty detached from the production config scenario. In the production case you can name them whatever you want and bring them in with the secret management system of choice.

Reply:
I thought about this, I just thought that you kind of get the feeling that the credentials file is then only supposed to be used on your local machine and maybe is just abusing the fact that we have *.local.yaml already in the `.gitignore`. I'm happy either way to be honest, just thought it was a little misleading.

Reply:
Mm I kinda get the same feeling

Reply:
Yeah agree, but tbh that's more or less the way it works. Those are your local credentials, and the ones you want to upload to secret management just happens to be the same? It would be valid to create a separate private key for use in the production configuration, iirc GitHub supports that.

Reply:
I created #4107 that adds `-credentials` to the file name and to gitignore. 
I'm happy to change it to .local if you have a strong preference.